### ISSUE TITLE
Add/expose random seed in LearningPipeline to get deterministic results

### ISSUE THREAD

**Issue by GalOshri** [dotnet/machinelearning#9]
It is important to be able to set a random seed for ML experiments so the results are reproducible. Add random seed to LearningPipeline (or elsewhere) to ensure a full experiment is deterministic.

**Comment by zeahmed** [comment]
I would suggest
``` C#
// Variant 3: Parameter to .Train()
var m = p.Train(seed:1234);
```

**Comment by Ivanidzo4ka** [comment]
so we have pipeline, we have transforms in it, we have

**Comment by TomFinley** [comment]
Seriously, enough is enough I think. Let's just expose `IHostEnvironment`. That's our mechanism to solve this and other problems, it's worked well for years doing exactly this.

**Comment by shauheen** [comment]
With a few PRs already merged as well as the proposed #371 , this issue can now be closed.
